tests/tt_metal/tt_metal/test_kernels/dataflow/unit_tests/command_queue/random_program.cpp

```diff
@@ -0,0 +1,43 @@
+
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#ifdef DATA_MOVEMENT
+#include <stdint.h>
+#include "dataflow_api.h"
+#endif
+
+#ifdef COMPUTE
+#include <cstdint>
+#include "compute_kernel_api/common.h"
+#include "compute_kernel_api/tile_move_copy.h"
+#include "compute_kernel_api/eltwise_unary/eltwise_unary.h"
+#include "compute_kernel_api/eltwise_unary/sfpu_split_includes.h"
+#endif
+
+#ifdef DATA_MOVEMENT
+namespace {
+void kernel_main() {
+#endif
+#ifdef COMPUTE
+namespace NAMESPACE {
+void MAIN  {
+#endif
+    constexpr volatile uint32_t outer_loop = get_compile_time_arg_val(0);
+    constexpr volatile uint32_t middle_loop = get_compile_time_arg_val(1);
+    constexpr volatile uint32_t inner_loop = get_compile_time_arg_val(2);
+
+    #pragma unroll(get_compile_time_arg_val(0))
+    for (volatile uint32_t i = 0; i < outer_loop; i++) {
+        #pragma unroll(get_compile_time_arg_val(1))
+        for (volatile uint32_t j = 0; j < middle_loop; j++) {
+            #pragma unroll(get_compile_time_arg_val(2))
+            for (volatile uint32_t k = 0; k < inner_loop; k++) {
+                // Do nothing
+            }
+        }
+    }
+}
```

Comment:
I assume the above is the method to get random sized kernels and perhaps random duration as well?
Note that test_pgm_dispatch uses a global variable and a timer to achieve that with precise control, fwiw.
Also, if you used data to make the program big, you could have a program which checks that validity of the data to be sure it all arrived correctly, though I guess executing is a pretty good sanity check (and initializing interesting patterns statically would be a challange)